spec.html

```diff
@@ -2,20 +2,490 @@
+          1. Return _asynccontextMapping_.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-asynccontext-constructor">
+      <h1>The AsyncContext Constructor</h1>
+      <p>The AsyncContext constructor:</p>
+      <ul>
+        <li>is <dfn>%AsyncContext%</dfn>.</li>
+        <li>is the initial value of the *"AsyncContext"* property of the global object.</li>
+        <li>creates and initializes a new AsyncContext when called as a constructor.</li>
+        <li>is not intended to be called as a function and will throw an exception when called in that manner.</li>
+        <li>may be used as the value in an `extends` clause of a class definition. Subclass constructors that intend to inherit the specified AsyncContext behaviour must include a `super` call to the AsyncContext constructor to create and initialize the subclass instance with the internal state necessary to support the `AsyncContext` and `AsyncContext.prototype` built-in methods.</li>
+      </ul>
+
+      <emu-clause id="sec-asynccontext">
+        <h1>AsyncContext ( )</h1>
+        <p>This function performs the following steps when called:</p>
+
+        <emu-alg>
+          1. If NewTarget is *undefined*, throw a *TypeError* exception.
+          1. Let _asynccontext_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%AsyncContext.prototype%"*, « [[AsyncContextKey]] »).
+          1. Set _asynccontext_.[[AsyncContextKey]] to a new Symbol.
+          1. Return _asynccontext_.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-properties-of-the-asynccontext-constructor">
+      <h1>Properties of the AsyncContext Constructor</h1>
+      <p>The AsyncContext constructor:</p>
+      <ul>
+        <li>has a [[Prototype]] internal slot whose value is %Function.prototype%.</li>
+        <li>has the following properties:</li>
+      </ul>
+
+      <emu-clause id="sec-asynccontext-wrap">
+        <h1>AsyncContext.wrap ( _func_ )</h1>
+        <p>This function returns a new function which opaquely captures the current value of all AsyncContexts and restores the captured value when being invoked.</p>
+
+        <emu-alg>
+          1. Let _snapshot_ be AsyncContextSnapshot().
+          1. Return ? AsyncContextWrappedFunctionCreate(_func_, _snapshot_).
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-properties-of-the-asynccontext-prototype-object">
+      <h1>Properties of the AsyncContext Prototype Object</h1>
+      <p>The <dfn>AsyncContext prototype object</dfn>:</p>
+      <ul>
+        <li>is <dfn>%AsyncContext.prototype%</dfn>.</li>
+        <li>has a [[Prototype]] internal slot whose value is %Object.prototype%.</li>
+        <li>is an ordinary object.</li>
+        <li>does not have any of the other internal slots of AsyncContext instances.</li>
+      </ul>
+
+      <emu-clause id="sec-asynccontext.prototype.run">
+        <h1>AsyncContext.prototype.run ( _value_, _func_ )</h1>
```

Comment:
I think we should include `…args` here, to match Node's behavior. It's not always needed, but it's convenient when I want to reuse a shared callback in different places and pass args to control the behavior.